### PR TITLE
Fixed geo_coordinate Decimal issue #253

### PR DESCRIPTION
**Pull request by wybiral**
Fixes issue #253 and makes the precision the same whether or not a center/radius is supplied.

### CODE REVIEW

File: faker/providers/address/__init__.py [hunk 1]
```diff
@@ -93,8 +93,10 @@ def geo_coordinate(cls, center=None, radius=0.001):
         if not center:
             return Decimal(str(random.randint(-180000000, 180000000) / 1000000.0)).quantize(Decimal('.000001'))
         else:
+            center = float(center)
+            radius = float(radius)
             geo = random.uniform(center - radius, center + radius)
```
Comment (fcurella) [inline faker/providers/address/__init__.py]: I'd rather cast `random.uniform` arguments than `center` and `radius`, to minimize precision loss:
```python
geo = random.uniform(float(center - radius), float(center + radius))
```

Comment (wybiral) [inline faker/providers/address/__init__.py]: Since radius defaults to floating point 0.001 in the function arguments it will break if all they supply is the center as a Decimal.

Comment (fcurella) [inline faker/providers/address/__init__.py]: Oh I see, good point.

This LGTM, Thank you! :)

**Comment by fcurella** [comment]
Thanks! :sparkles:

File: faker/providers/address/__init__.py (hunk 1 above)
Comment (wybiral) [inline faker/providers/address/__init__.py]: No problem.

It might also be a good idea to replace the check from ```if not center:``` to ```if center is None:``` in the event that someone supplies a center of 0 (otherwise it'll get randomized).

Comment (fcurella) [inline faker/providers/address/__init__.py]: Good call. Would you mind submit another PR?

Comment (wybiral) [inline faker/providers/address/__init__.py]: Sure thing, it's PR #281 

This is probably a bad place to ask (I'm new to Github), but does this method need some tests added since this seems to decrease the coverage/coveralls?

Comment (fcurella) [inline faker/providers/address/__init__.py]: We have some _basic_ testing, where every provider gets called at least once with the default arguments, but that obviously doesn't cover most cases.

New per-method tests would be a nice addition and are really appreciated.
